Classes/Domain/Model/Forum/Access.php

```diff
@@ -184,16 +186,16 @@ public function matches(FrontendUser $user = NULL) {
 				throw new \Exception('access record #' . $this->getUid() . ' is of login level type "specific", but has not valid affected user group', 1436527735);
 			}
 			if ($user !== NULL) {
-				foreach ($user->getUsergroup() as $group) {
-					/** @var $group \Mittwald\Typo3Forum\Domain\Model\User\FrontendUserGroup */
-					if ($group->getUid() === $this->affectedGroup->getUid()) {
+				$userGroupUids = GeneralUtility::trimExplode(',', $GLOBALS['TSFE']->gr_list);
```

Comment:
This changes the semantics of this function; up until now, the `matches` function could be used to check ACLs for *any given* user. Now, it will check only the groups of the *currently authenticated* user. Off the top of my head, I could think of a few different solutions for this:

1. Remove the `$user` parameter, to communicate explicitly that this methods checks only ACLs for the currently authenticated user. Also, this would break BC and technically, we would have to release a new major version after this.
2. Leave the `matches` method as-is, and add a new one named `matchesCurrentUser`.
3. Check where the `FrontendUser` instance that is passed into this method is built and adjust accordingly.
4. Add a new (optional) parameter like `$additionalGroupUids = []` that you can pass in addition to the `$user` object.

What do you think? I'd prefer option 3 or 4.